### PR TITLE
#RI-5416 - add rdi pipeline template

### PR DESCRIPTION
**Pull request by AmirAllayarovSofteq**

### CODE REVIEW

File: redisinsight/api/src/modules/rdi/client/rdi.client.ts [hunk 1]
```diff
@@ -23,6 +23,12 @@ export abstract class RdiClient {
 
   abstract getPipeline(): Promise<RdiPipeline>;
 
+  // TODO validate options and response
```
Comment (kyle-marcum) [inline redisinsight/api/src/modules/rdi/client/rdi.client.ts:26]: Is this going to be completed in a separate card?

File: redisinsight/api/src/modules/rdi/client/rdi.client.ts [hunk 2]
```diff
@@ -23,6 +23,12 @@ export abstract class RdiClient {
...
+  abstract getTemplate(options: object): Promise<unknown>;
+
+  // TODO validate response schema
```
Comment (kyle-marcum) [inline redisinsight/api/src/modules/rdi/client/rdi.client.ts:29]: Is this going to be completed in a separate card?

File: redisinsight/ui/src/slices/interfaces/rdi.ts [hunk 3]
```diff
@@ -1,11 +1,19 @@
 import { Nullable } from 'uiSrc/utils'
 import { Rdi as RdiInstanceResponse } from 'apiSrc/modules/rdi/models/rdi'
+import {PageNames} from "uiSrc/constants";
```
Comment (kyle-marcum) [inline redisinsight/ui/src/slices/interfaces/rdi.ts:3]: Screenshot 2024-03-01 at 3 30 49 PM

File: redisinsight/api/src/modules/rdi/client/rdi.client.ts (hunk 1 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/api/src/modules/rdi/client/rdi.client.ts:26]: Yep, i guess in integration card (i'm asked rdi about request and response, but didn't get answer)

File: redisinsight/api/src/modules/rdi/client/rdi.client.ts (hunk 2 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/api/src/modules/rdi/client/rdi.client.ts:29]: same

File: redisinsight/ui/src/slices/interfaces/rdi.ts (hunk 3 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/ui/src/slices/interfaces/rdi.ts:3]: Thank you. Fixed

File: redisinsight/ui/src/slices/rdi/pipeline.ts [hunk 4]
```diff
@@ -119,6 +149,60 @@ export function deployPipelineAction(
...
+
+      dispatch(getPipelineStrategiesFailure(parsedError))
+      onFailAction?.()
+    }
+  }
+}
+
+export function fetchPipelineTemplate(
+  rdiInstanceId: string,
+  // TODO add interface
```
Comment (kyle-marcum) [inline redisinsight/ui/src/slices/rdi/pipeline.ts:181]: Is this going to be completed in a separate card?

File: redisinsight/ui/src/slices/tests/rdi/pipeline.spec.ts [hunk 5]
```diff
@@ -10,11 +10,15 @@ import reducer, {
...
+  getPipelineStrategies,
+  getPipelineStrategiesSuccess,
+  getPipelineStrategiesFailure,
+  setPipelineSchema,
   fetchRdiPipeline,
   deployPipelineAction,
   rdiPipelineSelector,
   fetchRdiPipelineSchema,
-  setPipelineSchema,
+  fetchPipelineStrategies, fetchPipelineTemplate,
```
Comment (kyle-marcum) [inline redisinsight/ui/src/slices/tests/rdi/pipeline.spec.ts:21]: small nitpick - could you put these on two separate lines?

File: redisinsight/ui/src/slices/rdi/pipeline.ts (hunk 4 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/ui/src/slices/rdi/pipeline.ts:181]: Yep. the same. In rdi docs right now we have template for job and for config in one request. But in our realization it doesn't make sense.
I will take a look at the bug. Which you found

File: redisinsight/ui/src/slices/tests/rdi/pipeline.spec.ts (hunk 5 above)
Comment (AmirAllayarovSofteq) [reply redisinsight/ui/src/slices/tests/rdi/pipeline.spec.ts:21]: For sure. done :)

**Review by kyle-marcum — APPROVED** [review]

**Pull request merged**